### DISCUSSION TITLE
How to display recursive size of directories in linemode

### DISCUSSION THREAD

**Discussion by fritzrehde** [sxyazi/yazi#492] [1. Q&A] [answered]
I know this is feature is probably discouraged due to performance bottlenecks, but I would really like a way to see the recursive disk usage of entire directories. I was wondering if I could somehow use the new plugin system to make this work. Would it be possible to display the size of files *and* directories in the linemode somehow?

**Comment by sxyazi** [comment, marked as answer]
In fact it's already supported at the moment, but deliberately restricted in lazily calculating directory sizes when sorting by size.

This means that you can switch the sorting method to size by pressing `,s`, and then you'll notice directory sizes appearing in linemode.

**Reply by fritzrehde** [reply]
Awesome! This is truly an amazing feature, and I really like the decision to only show recursive directory sizes when sorting by size. Great work!

**Reply by ntlv1** [reply]
Thanks. I'm looking for a way to show size of the directory under cursor using a hotkey. Is it possible?

**Reply by sxyazi** [reply]
Currently not supported, but we can add it—perhaps as a standalone command or by adding relevant APIs to allow users to implement it through the plugin system.

**Reply by og900aero** [reply]
It would also be a good solution to show the size of a directory when you select it (such as when you want to copy or move it). In fact. In the case of several such selections, the sum of the sizes of the selected elements would also be shown in the status line, so it is easy to know whether what we are copying/moving will fit on the given medium, etc.
